src/Chapter1.hs

```diff
@@ -554,7 +556,12 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+  |  y <= x && x <= z || y >= x && x >= z = x
+  |  x <= y && y <= z || x >= y && y >= z = y
+  |  x <= z && z <= y || x >= z && z >= y = z
+  | otherwise = x
```

Comment:
Yeah.
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive`, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

Because of that, you needed to add another guard – `| otherwise = x`, to tell the compiler, that your pattern matching is exhaustive 🙂
However, you have already covered all possible option, so it is not necessary to check the conditions for `z` and just return `z`

```suggestion
  | otherwise = z
```